000-vip-init.php

```diff
@@ -9,6 +9,24 @@
  * good reason not to.
  */
 
+if ( defined( 'VIP_GO_SITE_BLOCKED' ) && VIP_GO_SITE_BLOCKED ) {
+	if ( '/cache-healthcheck?' === $_SERVER['REQUEST_URI'] ) {
+		if ( function_exists( 'newrelic_end_transaction' ) ) {
+			// Discard the transaction (the `true` param)
+			// See: https://docs.newrelic.com/docs/agents/php-agent/configuration/php-agent-api#api-end-txn
+			newrelic_end_transaction( true );
+		}
+
+		http_response_code( 200 );
+
+		die( 'ok (site blocked)' );
+	}
+
+	http_response_code( 503 );
+
+	die( 'Site is currently under maintenance' );
```

Comment:
We should probably have a nice version of the maintenance page like this one: https://github.com/Automattic/vip-go-mu-plugins/blob/master/errors/alloptions-limit.html

Reply:
Good call 👍